sdk/src/lib.rs

```diff
@@ -106,24 +197,36 @@ pub use solana_account as account;
     note = "Use `solana_account::state_traits` crate instead"
 )]
 pub use solana_account::state_traits as account_utils;
-#[deprecated(since = "2.1.0", note = "Use `solana-bn254` crate instead")]
-pub use solana_bn254 as alt_bn128;
+#[deprecated(since = "2.3.0", note = "Use `solana-bn254` crate instead")]
+pub mod alt_bn128 {
+    pub use solana_bn254::*;
+}
 #[cfg(feature = "full")]
-#[deprecated(since = "2.2.0", note = "Use `solana-client-traits` crate instead")]
-pub use solana_client_traits as client;
+#[deprecated(since = "2.3.0", note = "Use `solana-client-traits` crate instead")]
+pub mod client {
+    pub use solana_client_traits::*;
+}
 #[deprecated(
-    since = "2.2.0",
+    since = "2.3.0",
     note = "Use `solana-compute-budget-interface` crate instead"
 )]
 #[cfg(feature = "full")]
-pub use solana_compute_budget_interface as compute_budget;
-#[deprecated(since = "2.1.0", note = "Use `solana-decode-error` crate instead")]
-pub use solana_decode_error as decode_error;
-#[deprecated(since = "2.1.0", note = "Use `solana-derivation-path` crate instead")]
-pub use solana_derivation_path as derivation_path;
+pub mod compute_budget {
+    pub use solana_compute_budget_interface::*;
+}
+#[deprecated(since = "2.3.0", note = "Use `solana-decode-error` crate instead")]
+pub mod decode_error {
+    pub use solana_decode_error::*;
+}
+#[deprecated(since = "2.3.0", note = "Use `solana-derivation-path` crate instead")]
+pub mod derivation_path {
+    pub use solana_derivation_path::*;
+}
 #[cfg(feature = "full")]
-#[deprecated(since = "2.2.0", note = "Use `solana-ed25519-program` crate instead")]
-pub use solana_ed25519_program as ed25519_instruction;
+#[deprecated(since = "2.3.0", note = "Use `solana-ed25519-program` crate instead")]
```

Comment:
why is there no `solana-ed25519-interface`?

Reply:
i think precompiles were a mistake. proposed solution will do. just annoying how many similar crate naming patterns contain inconsistent symbol paths